internal/artifact/artifact.go

```diff
@@ -78,20 +82,30 @@ func (a Artifact) ExtraOr(key string, or interface{}) interface{} {
 	return a.Extra[key]
 }
 
-// Checksum calculates the SHA256 checksum of the artifact.
-func (a Artifact) Checksum() (string, error) {
-	log.Debugf("calculating sha256sum for %s", a.Path)
+// Checksum calculates the checksum of the artifact.
+func (a Artifact) Checksum(algorithm string) (string, error) {
+	log.Debugf("calculating checksum for %s", a.Path)
 	file, err := os.Open(a.Path)
 	if err != nil {
 		return "", errors.Wrap(err, "failed to checksum")
 	}
 	defer file.Close() // nolint: errcheck
-	var hash = sha256.New()
-	_, err = io.Copy(hash, file)
+	var h hash.Hash
+	switch algorithm {
```

Comment:
```suggestion
	switch algorithm {
		case "crc32":
			hash = crc32.NewIEEE()
		case "md5":
			hash = md5.New()
		case "sha224":
			hash = sha256.New224()
		case "sha384":
			hash = sha512.New384()
```

Might as well add some of the other common types as well. I realize the use case is minimal, but it's almost 0 cost.

Reply:
yeah, why not haha